### PR TITLE
Exposes the map instance

### PR DESCRIPTION
**Pull request by aditudorache**
This PR exposes the map instance to the parent (#88 )

### CODE REVIEW

File: src/Map.tsx [hunk 1]
```diff
@@ -26,7 +28,12 @@ const Map: React.FC<React.HTMLProps<HTMLDivElement> & MapProps> = ({
 
   useEffect(() => {
     if (mapRef.current !== null) {
-      setMapInstance(new L.Map(mapRef.current, options))
+      const map = new L.Map(mapRef.current, options)
```
Comment (jonkoops) [inline src/Map.tsx]: I think this should be done conditionally if `setInstance` is not passed.

File: src/Map.tsx [hunk 2]
```diff
@@ -10,12 +10,14 @@ import useEvents from './utils/useEvents'
 export interface MapProps {
   events?: LeafletEventHandlerFnMap
   options?: MapOptions
+  setInstance?: (instance: L.Map) => void
```
Comment (jonkoops) [inline src/Map.tsx]: Instead of passing `setInstance` would it not be better to pass `mapInstance` as a prop instead?

**Comment by jonkoops** [comment]
I think it should already be possible to get the map from a parent component but I am not quite sure how. @chrisvanmook could you take a look at this as well?

File: src/Map.tsx (hunk 1 above)
Comment (aditudorache) [reply src/Map.tsx]: I wouldn't do that , the Map component is responsible for instancing the L.Map. In some cases a pan or other state action must be triggered from outside the map.

File: src/Map.tsx (hunk 2 above)
Comment (aditudorache) [reply src/Map.tsx]: I wouldn't do that, the Map component should be the only responsible for creating a L.Map object

**Comment by aditudorache** [comment]
@chrisvanmook it is required sometimes to change the zoom/pan as a consequence of changes from outside the application. Now we can move the marker from outside the map but we cannot pan to that.

File: src/Map.tsx (hunk 1 above)
Comment (jonkoops) [reply src/Map.tsx]: Ah, yes. I see it's just passing the map instance along. That's fine.

**Pull request merged**